index.rb

```diff
@@ -19,56 +21,106 @@ def my_select
     result
   end
 
-  def my_all?
-    my_each do |e|
-      return false unless yield(e)
+  def my_all?(*args)
+    if !args[0].nil?
+      my_each { |item| return false unless args[0] == item }
+    elsif block_given?
+      my_each { |item| return false unless yield(item) }
+    elsif args.instance_of?(Class)
+      my_each { |item| return false unless yield(item) != args }
+    elsif args.instance_of?(Regexp)
+      my_each { |item| return false unless args.match(item) }
+    else
+      my_each { |item| return false unless item }
```

Comment:
- [ ] The method doesn't pass the test for Classes and boolean cases .e.g 
```ruby
[1, 2i, 3.14].my_all?(Numeric)         #returned false instead of true
[nil, true, 99].my_all?                       #returned true instead of false
```